webpack/assets/javascripts/services/ChartService.js

```diff
@@ -86,6 +90,40 @@ export const getDonutChartConfig = ({
     id,
   });
 
+export const getBarChartConfig = ({
+  data,
+  config,
+  onclick,
+  label,
+  id = uuidV1(),
+}) => {
+  const chartConfig = getChartConfig({
+    data,
+    config,
+    onclick,
+    id,
+  });
+
+  const categories = data.map(dataItem => dataItem[0]);
+
+  const columns = data.map(x => x[1]);
```

Comment:
@sharvit This `ChartSevice` becomes more and more complex, each render this will invoked, even if nothing changed.
what do you think about moving this logic to a selector (memorized if needed) by connect this component to the store, or its parent.

Reply:
Can we actually move it into selectors? I think most of the data coming from the data prop and not from the state.
Does it actually lead to performance issue that we must solve in this particular PR?